src/App.tsx

```diff
@@ -1,8 +1,115 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
+import './App.scss';
+import {
+  getComments, getPosts, getUsers, Posts,
+} from './helpers/api';
+import { LoadingSpinner } from './loading-spinner';
 
-const App = () => (
-  <h1>Dynamic list of posts</h1>
-);
+const App = () => {
+  const [post, setPost] = useState<Posts[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [sortedPosts, setSortedPosts] = useState<Posts[]>([]);
+
+  const handleLoad = async () => {
+    setIsLoading(true);
+    try {
+      const postsFromServer = await getPosts();
+      const usersFromServer = await getUsers();
+      const commentsFromServer = await getComments();
+
+      const allPosts = postsFromServer.map(newPost => ({
+        ...newPost,
+        user: usersFromServer.find(user => user.id === newPost.userId),
+        comments: commentsFromServer.filter(comment => comment.postId === newPost.id),
+      }));
+
+      setPost(allPosts);
+      setIsLoaded(true);
+      setSortedPosts(allPosts);
+    } catch (e) {
+      setErrorMessage('Loading from server terminated. Please try again later');
+    }
+
+    setIsLoading(false);
+  };
+
+  const sortPosts = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
just return a new array here and assign it to the array you need in render

Reply:
resolved